### PR TITLE
David

### PR DESCRIPTION
**Pull request by schwiti6190**

### CODE REVIEW

File: BunkersiloManager.lua [hunk 1]
```diff
@@ -9,42 +35,54 @@ end
...
-	local sy = getTerrainHeightAtWorldPos(g_currentMission.terrainRootNode, sx, 1, sz);
 	local bunkerWidth = courseplay:distance(sx,sz, wx, wz)
 	local bunkerLength = courseplay:distance(sx,sz, hx, hz)
+	local sy = getTerrainHeightAtWorldPos(g_currentMission.terrainRootNode, sx, 1, sz);
 	local startDistance = courseplay:distanceToPoint(vehicle, sx, sy, sz)
 	local endDistance = courseplay:distanceToPoint(vehicle, hx, sy, hz)
+	
+	--correct data for bunkerSilos
+	if Silo.bunkerSiloArea.start then
+		sx, _, sz = localToWorld(Silo.bunkerSiloArea.start,-0.5,0,0) --start BunkerNode
```
Comment (pvaiko) [inline BunkersiloManager.lua:57]: a comment would be nice explaining what the purpose of these adjustments are...

File: LevelCompactAIDriver.lua [hunk 2]
```diff
@@ -58,38 +56,36 @@ function LevelCompactAIDriver:init(vehicle)
 	self:setHudContent()
 	self.fillUpState = self.states.PUSH
 	self.stoppedCourseplayers = {}
-	self:getWorkWidth()
+	self:setLevelerWorkWidth()
+	self.workWidth = 3
```
Comment (pvaiko) [inline LevelCompactAIDriver.lua:60]: isn't this overwriting what the line above just calculated?

File: CpManager.lua [hunk 3]
```diff
@@ -829,13 +829,15 @@ end;
...
 
+--FieldScan startup dialog and github disclaimer
 function CpManager:showYesNoDialogue(title, text, callbackFn)
 	-- don't show anything if the tutorial dialog is open (it takes a while until is isOpen shows true after startup, hence the clock)
 	--courseplay.debugFormat(12, "clock %d %s", courseplay.clock, tostring(g_gui.guis.YesNoDialog.target and g_gui.guis.YesNoDialog.target.isOpen))
 	if courseplay.clock < 2000 or (g_gui.guis.YesNoDialog.target and g_gui.guis.YesNoDialog.target.isOpen) then
 		return
 	end
 	--courseplay.debugFormat(12, text)
+	local text =string.format("%s\n %s",courseplay:loc('COURSEPLAY_YES_NO_FIELDSCAN'),courseplay:loc('COURSEPLAY_YES_NO_FIELDSCAN_DISCLAIMER'))
```
Comment (pvaiko) [inline CpManager.lua:840]: Could we rename COURSEPLAY_YES_NO_FIELDSCAN_DISCLAIMER to COURSEPLAY_SUPPORT_INFO or something, the text has nothing to do with field scanning (we just put it into the same dialog) and isn't really a disclaimer either :)

**Review by pvaiko — CHANGES_REQUESTED** [review]

**Comment by Redrooster1-web** [comment]
I Apprieciate all what you are doing, all of you. But how do i setup a c/p course to lvl  chaff/grass??????????

E-mail me at ajboyle873@gmail.com

**Pull request merged**